src/Collections-Unordered/Dictionary.class.st

```diff
@@ -85,6 +85,15 @@ Dictionary class >> newFrom: aDict [
 "
 ]
 
+{ #category : #'instance creation' }
+Dictionary class >> newFromAssociations: anAssocCollection [
+	"Return a dictionary with the associations defined in anAssocCollection. If the same key appears twice in anAssocCollection, the last one enumerated win."
+	
+	^ anAssocCollection 
+		inject: self new 
+		into: [ :dict :assoc | dict add: assoc. dict ]
```

Comment:
Maybe it's just me. But I found it quite complicated. Why not use simple version with do?
```Smalltalk
result := Dictionary new: anAssocCollection size.
 anAssocCollection do: [:each | result add: each ].
^result
```

Reply:
Tx denis.
I followed the implementation of kasper. Now I should bench them to see which one is faster and pick the fastest one.